### PR TITLE
add support for sslHandshakeTimeout in NetServer and NetClient Options

### PR DESCRIPTION
**Pull request by BillyYccc**
Fix Issue #2118 
add implement and some unit tests for sslHandshakeTimeout in NetClient Options
add an API to expose netty's SSL Handshake Timeout
Signed-off-by: BillyYccc <billy112487983@gmail.com>

### CODE REVIEW

**Comment by vietj** [comment]
10 seconds is a big long for this test

**Comment by vietj** [comment]
I'm wondering if that's not conflicting with the current default idle timeout, i.e the idle timeout set before the handshake could fill the same purpose

**Comment by vietj** [comment]
also perhaps if we accept it, it should also apply not only to `NetClient`

File: src/test/java/io/vertx/test/core/NetTest.java [hunk 1]
```diff
@@ -1261,6 +1288,20 @@ public void testSpecificTlsProtocolVersion() throws Exception {
   }
 
   @Test
+  // test client SSL handshakeTimeout option
+  public void testClientSslHandshakeTimeout() throws Exception {
+    testTLS(Cert.NONE, Trust.NONE, Cert.SERVER_JKS, Trust.NONE, false, true, true, false, new String[0],
```
Comment (vietj) [inline src/test/java/io/vertx/test/core/NetTest.java]: I think SSL timeout should have specific test and not reuse the general one, for instance for server:

- setup an SSL server with a timeout with a small value for instance 400ms
- run a NetClient that connects to the server and does nothing
- wait for the SSL timeout to fail, i.e the client will get an error on connect

Comment (BillyYccc) [reply src/test/java/io/vertx/test/core/NetTest.java]: Thanks for your helpful suggestions! I'm working on it

**Comment by BillyYccc** [comment]
@vietj I think I'm ready now, but current version of netty does not support the setHandshakeTimeout in server side, see issue in **Netty** [issue #7230](https://github.com/netty/netty/issues/7230). We may need to wait for the fix of that issue in **Netty** and upgrade our dependency to let this make sense.

**Comment by vietj** [comment]
@BillyYccc yes, I've followed this issue on the Netty side as well

**Comment by BillyYccc** [comment]
Hi! After [issue #7230](https://github.com/netty/netty/issues/7230) of Netty has been fixed in Netty 4.1.17.Final. Recently Vert.x has upgraded netty dependency to 4.1.19. So this issue can be scheduled now. I have rewritten the commits and messages to make it cleaner.

**Comment by BillyYccc** [comment]
Duplicate of https://github.com/eclipse-vertx/vert.x/pull/2714.

**Pull request closed without merging**